Change reference cost year to 2020

lisazeyen opened the pull request:
Closes # (if applicable).

## Changes proposed in this Pull Request

Change the reference cost year to 2020, closes issue #118 

- a new column `currency_year` is added which represents the currency year of the original input data
- all input data is adjusted for inflation
- for input data where currency year was not clear, year of publication is assumed
- for old pypsa cost assumptions currency year 2015 is assumed
## Checklist


- [x] add annual inflation from [eurostat](https://ec.europa.eu/eurostat/databrowser/bookmark/8ee76f19-0146-4570-991a-0934ebd6a628?lang=de)
- [ ] @JulianGeis one would need to go through all the techs listed [here](https://github.com/PyPSA/technology-data/blob/df3632810302f57c97af45d9fffdc4225845f3b6/scripts/compile_cost_assumptions.py#L65-L139) and double check the reference cost year. All the techs with cost year 2020 should be added to the list [here](https://github.com/PyPSA/technology-data/blob/df3632810302f57c97af45d9fffdc4225845f3b6/scripts/compile_cost_assumptions.py#L209-L214)
- [x] add reference year for vehicle costs

- [x] Code changes are sufficiently documented; i.e. new functions contain docstrings and further explanations may be given in `doc`.
- [x] Data source for new technologies is cleary stated.
- [x] Newly introduced dependencies are added to `environment.yaml` (if applicable).
- [x] A note for the release notes `doc/release_notes.rst` of the upcoming release is included.
- [x] I consent to the release of this PR's code under the GPLv3 license.

fneum commented:
@Aleks-g and @koen-vg have used real inflation data instead of `snakemake.config['rate_inflation']` from https://ec.europa.eu/eurostat/databrowser/view/teicp000/default/table?lang=en in https://github.com/koen-vg/enabling-agency / https://arxiv.org/abs/2312.11264. 

>  While costs in the above repository are given in 2015 EUR, we have converted all cost data to 2023 EUR for the purposes of this study, using inflation data for the Euro area up to October 2023 [13], amounting to a 24.5% increase compared to technology-data.

Don't know though if that was done afterwards or directly in a fork of `technology-data`.

aleks-g commented:
> @aleks-g and @koen-vg have used real inflation data instead of `snakemake.config['rate_inflation']` from https://ec.europa.eu/eurostat/databrowser/view/teicp000/default/table?lang=en in https://github.com/koen-vg/enabling-agency / https://arxiv.org/abs/2312.11264.
> 
> > While costs in the above repository are given in 2015 EUR, we have converted all cost data to 2023 EUR for the purposes of this study, using inflation data for the Euro area up to October 2023 [13], amounting to a 24.5% increase compared to technology-data.
> 
> Don't know though if that was done afterwards or directly in a fork of `technology-data`.

Hi Fabian, we've only included inflation since 2015 in the analysis afterwards, without adapting anything in technology-data itself. :)

lisazeyen commented:
> @aleks-g and @koen-vg have used real inflation data instead of `snakemake.config['rate_inflation']` from https://ec.europa.eu/eurostat/databrowser/view/teicp000/default/table?lang=en in https://github.com/koen-vg/enabling-agency / https://arxiv.org/abs/2312.11264.
> 
> > While costs in the above repository are given in 2015 EUR, we have converted all cost data to 2023 EUR for the purposes of this study, using inflation data for the Euro area up to October 2023 [13], amounting to a 24.5% increase compared to technology-data.
> 
> Don't know though if that was done afterwards or directly in a fork of `technology-data`.

I guess it make sense to take the actual inflation. I downloaded the annual inflation from eurostat and would add this instead of assuming 2% inflation for every year

lisazeyen commented:
We haven't checked all technologies yet in DEA, but I think we can already pull it in and open another issue that all single DEA technologies should be double checked for the correct currency year.

koen-vg commented:
I have to say I'm a little bit confused by the Eurostat dataset browser sometimes because it sometimes includes and sometimes doesn't include 2023. But here it seems possible to get a dataset in the same format with 2023 included? https://ec.europa.eu/eurostat/databrowser/view/prc_hicp_aind__custom_9900786/default/table?lang=en

I think it would be nice to have as recent as possible data available, and there are now inflation values for 2023 (namely, 6.4%).

fneum commented:
Let's do the following spot checks:

- [ ] H2 (g) pipeline
- [ ] HVDC overhead
- [ ] nuclear
- [ ] methanolisation

lisazeyen commented on docs/release_notes.rst:
```diff
@@ -5,6 +5,8 @@ Release Notes
 .. Upcoming Release
 .. ================
 
+* Update currency year from 2015 to 2020. Add a currency year for each data input
```
I compared for all technologies the original data from the data source and the values after the adjustment for the inflation. Will do the suggested spot checks as well [y-axis = after inflation adjustment/ before inflation adjustment]
![before_after_currency_year](https://github.com/PyPSA/technology-data/assets/35347358/10a5d941-36b2-4f28-a3f5-895b90a06985)

lisazeyen commented on environment.yaml:
```diff
@@ -3,15 +3,15 @@ channels:
   - conda-forge
   - bioconda
 dependencies:
-  - python>=3.9
```
I couldn't run snakemake workflow with python 3.12 so I restricted python for now between 3.9-3.11

lisazeyen commented:
> Let's do the following spot checks:
> 
> * [ ]  H2 (g) pipeline
> * [ ]  HVDC overhead
> * [ ]  nuclear
> * [ ]  methanolisation

H2 (g) pipeline, investment cost of currency year 2023

```
                             old       new
2020 FOM                  4.0000    4.0000
     electricity-input    0.0210    0.0210
     investment         309.8853  320.8742
     lifetime            50.0000   50.0000
2025 FOM                  3.5833    3.5833
     electricity-input    0.0200    0.0200
     investment         309.8853  320.8742
     lifetime            50.0000   50.0000
2030 FOM                  3.1667    3.1667
     electricity-input    0.0190    0.0190
     investment         309.8853  320.8742
     lifetime            50.0000   50.0000
2035 FOM                  2.7500    2.7500
     electricity-input    0.0185    0.0185
     investment         309.8853  320.8742
     lifetime            50.0000   50.0000
2040 FOM                  2.3333    2.3333
     electricity-input    0.0180    0.0180
     investment         309.8853  320.8742
     lifetime            50.0000   50.0000
2045 FOM                  1.9167    1.9167
     electricity-input    0.0175    0.0175
     investment         309.8853  320.8742
     lifetime            50.0000   50.0000
2050 FOM                  1.5000    1.5000
     electricity-input    0.0170    0.0170
     investment         309.8853  320.8742
     lifetime            50.0000   50.0000
```

HVDC overhead, currency_year 2011
```
                      old       new
2020 FOM           2.0000    2.0000
     investment  432.9729  451.8009
     lifetime     40.0000   40.0000
2025 FOM           2.0000    2.0000
     investment  432.9729  451.8009
     lifetime     40.0000   40.0000
2030 FOM           2.0000    2.0000
     investment  432.9729  451.8009
     lifetime     40.0000   40.0000
2035 FOM           2.0000    2.0000
     investment  432.9729  451.8009
     lifetime     40.0000   40.0000
2040 FOM           2.0000    2.0000
     investment  432.9729  451.8009
     lifetime     40.0000   40.0000
2045 FOM           2.0000    2.0000
     investment  432.9729  451.8009
     lifetime     40.0000   40.0000
2050 FOM           2.0000    2.0000
     investment  432.9729  451.8009
     lifetime     40.0000   40.0000
```

nuclear, currency year investment 2023, fuel 2010

```
old        new
2020 FOM            1.2700     1.2700
     VOM            3.6188     3.7471
     efficiency     0.3260     0.3260
     fuel           3.3122     3.4495
     investment  8769.6136  9080.5958
     lifetime      40.0000    40.0000
2025 FOM            1.2700     1.2700
     VOM            3.6188     3.7471
     efficiency     0.3260     0.3260
     fuel           3.3122     3.4495
     investment  8769.6136  9080.5958
     lifetime      40.0000    40.0000
2030 FOM            1.2700     1.2700
     VOM            3.6188     3.7471
     efficiency     0.3260     0.3260
     fuel           3.3122     3.4495
     investment  8769.6136  9080.5958
     lifetime      40.0000    40.0000
2035 FOM            1.2700     1.2700
     VOM            3.6188     3.7471
     efficiency     0.3260     0.3260
     fuel           3.3122     3.4495
     investment  8769.6136  9080.5958
     lifetime      40.0000    40.0000
2040 FOM            1.2700     1.2700
     VOM            3.6188     3.7471
     efficiency     0.3260     0.3260
     fuel           3.3122     3.4495
     investment  8769.6136  9080.5958
     lifetime      40.0000    40.0000
2045 FOM            1.2700     1.2700
     VOM            3.6188     3.7471
     efficiency     0.3260     0.3260
     fuel           3.3122     3.4495
     investment  8769.6136  9080.5958
     lifetime      40.0000    40.0000
2050 FOM            1.2700     1.2700
     VOM            3.6188     3.7471
     efficiency     0.3260     0.3260
     fuel           3.3122     3.4495
     investment  8769.6136  9080.5958
     lifetime      40.0000    40.0000
```
     
     methanolisation, currency year 2017
     
  ```
                                       old          new
2020 FOM                       3.0000       3.0000
     capture rate              0.9000       0.9000
     carbondioxide-input       0.2480       0.2480
     electricity-input         0.2710       0.2710
     heat-output               0.1000       0.1000
     hydrogen-input            1.1380       1.1380
     investment           757400.9996  826429.2092
     lifetime                 20.0000      20.0000
2025 FOM                       3.0000       3.0000
     capture rate              0.9000       0.9000
     carbondioxide-input       0.2480       0.2480
     electricity-input         0.2710       0.2710
     heat-output               0.1000       0.1000
     hydrogen-input            1.1380       1.1380
     investment           704056.1323  768222.5834
     lifetime                 20.0000      20.0000
2030 FOM                       3.0000       3.0000
     capture rate              0.9000       0.9000
     carbondioxide-input       0.2480       0.2480
     electricity-input         0.2710       0.2710
     heat-output               0.1000       0.1000
     hydrogen-input            1.1380       1.1380
     investment           650711.2649  710015.9577
     lifetime                 20.0000      20.0000
2035 FOM                       3.0000       3.0000
     capture rate              0.9000       0.9000
     carbondioxide-input       0.2480       0.2480
     electricity-input         0.2710       0.2710
     heat-output               0.1000       0.1000
     hydrogen-input            1.1380       1.1380
     investment           608179.5463  663607.9722
     lifetime                 20.0000      20.0000
2040 FOM                       3.0000       3.0000
     capture rate              0.9000       0.9000
     carbondioxide-input       0.2480       0.2480
     electricity-input         0.2710       0.2710
     heat-output               0.1000       0.1000
     hydrogen-input            1.1380       1.1380
     investment           565647.8278  617199.9868
     lifetime                 20.0000      20.0000
2045 FOM                       3.0000       3.0000
     capture rate              0.9000       0.9000
     carbondioxide-input       0.2480       0.2480
     electricity-input         0.2710       0.2710
     heat-output               0.1000       0.1000
     hydrogen-input            1.1380       1.1380
     investment           523116.1092  570792.0014
     lifetime                 20.0000      20.0000
2050 FOM                       3.0000       3.0000
     capture rate              0.9000       0.9000
     carbondioxide-input       0.2480       0.2480
     electricity-input         0.2710       0.2710
     heat-output               0.1000       0.1000
     hydrogen-input            1.1380       1.1380
     investment           480584.3906  524384.0160
     lifetime                 20.0000      20.0000
     ```

euronion commented:
> > I have to say I'm a little bit confused by the Eurostat dataset browser sometimes because it sometimes includes and sometimes doesn't include 2023. But here it seems possible to get a dataset in the same format with 2023 included? https://ec.europa.eu/eurostat/databrowser/view/prc_hicp_aind__custom_9900786/default/table?lang=en
> > I think it would be nice to have as recent as possible data available, and there are now inflation values for 2023 (namely, 6.4%).
> > Apart from that, thanks a bunch for this work!
> 
> Good idea to include 2023, unfortunately the data you are linking to is only dating back to 2015 and it would be nice to have the earlier years in there as well. I have added 2023 now within the function when preparing the data (not the nicest way, but at least 2023 is included now as well)...

You could use the OECD data repository instead, they list HICP since 1991 for the current EURO area and allow for download as CSV or by API access:

https://data-explorer.oecd.org/vis?tm=HICP&pg=0&snb=5&vw=tb&df[ds]=dsDisseminateFinalDMZ&df[id]=DSD_PRICES%40DF_PRICES_HICP&df[ag]=OECD.SDD.TPS&df[vs]=1.0&pd=%2C&dq=.A.HICP.CPI.PA._T.N.GY&ly[rw]=REF_AREA&ly[cl]=TIME_PERIOD&to[TIME_PERIOD]=false&lo=100&lom=LASTNPERIODS

lisazeyen commented:
> > > I have to say I'm a little bit confused by the Eurostat dataset browser sometimes because it sometimes includes and sometimes doesn't include 2023. But here it seems possible to get a dataset in the same format with 2023 included? https://ec.europa.eu/eurostat/databrowser/view/prc_hicp_aind__custom_9900786/default/table?lang=en
> > > I think it would be nice to have as recent as possible data available, and there are now inflation values for 2023 (namely, 6.4%).
> > > Apart from that, thanks a bunch for this work!
> > 
> > 
> > Good idea to include 2023, unfortunately the data you are linking to is only dating back to 2015 and it would be nice to have the earlier years in there as well. I have added 2023 now within the function when preparing the data (not the nicest way, but at least 2023 is included now as well)...
> 
> You could use the OECD data repository instead, they list HICP since 1991 for the current EURO area and allow for download as CSV or by API access:
> 
> [https://data-explorer.oecd.org/vis?tm=HICP&pg=0&snb=5&vw=tb&df[ds]=dsDisseminateFinalDMZ&df[id]=DSD_PRICES%40DF_PRICES_HICP&df[ag]=OECD.SDD.TPS&df[vs]=1.0&pd=%2C&dq=.A.HICP.CPI.PA._T.N.GY&ly[rw]=REF_AREA&ly[cl]=TIME_PERIOD&to[TIME_PERIOD]=false&lo=100&lom=LASTNPERIODS](https://data-explorer.oecd.org/vis?tm=HICP&pg=0&snb=5&vw=tb&df%5Bds%5D=dsDisseminateFinalDMZ&df%5Bid%5D=DSD_PRICES%40DF_PRICES_HICP&df%5Bag%5D=OECD.SDD.TPS&df%5Bvs%5D=1.0&pd=%2C&dq=.A.HICP.CPI.PA._T.N.GY&ly%5Brw%5D=REF_AREA&ly%5Bcl%5D=TIME_PERIOD&to%5BTIME_PERIOD%5D=false&lo=100&lom=LASTNPERIODS)

they are mentioning as source Eurostat (=.
I found a Eurostat table now with data from 2001-2023 and added that one as well as the API link to the source

lisazeyen commented:
@fneum I compared now the original data input with the cost adjusted values. Comparing the old and new costs is not very insightful since many of the input data previously was not inflation adjusted.

One thing to double check, I am basically assuming now when converting e.g. Eur_2017 to Eur_2020 the inflation of the three year 2018, 2019, 2020, since Eur_2017 should consider already the inflation for 2017 right?

fneum commented:
> One thing to double check, I am basically assuming now when converting e.g. Eur_2017 to Eur_2020 the inflation of the three year 2018, 2019, 2020, since Eur_2017 should consider already the inflation for 2017 right?

I think either way is fine, just it cannot be 2017, 2018, 2019, 2020.
